### DISCUSSION TITLE
PyBaMM version and Temperature at each time step in a DFN model with thermal submodel

### DISCUSSION THREAD

**Discussion by abhishek-appana** [pybamm-team/PyBaMM#3819] [Q&A] [answered]
Hello everyone,

I am new to PyBaMM and I am still trying to figure things out! I appreciate the help, in advance.

I recently inherited a model from another team who were using an earlier version of PyBaMM. Building on that, I am working on putting together a DFN model which has various submodels in it. One of them is the thermal submodel (2+1D current collectors thermal submodel).

I have 2 questions:
- The PyBaMM source directory that I have received doesn't have the "version.py" file in it. It has a file named "version" and the file type is shown as "file". On opening it, it has the numbers "0, 4, 0" written in it. How do I figure out which PyBaMM version was being used?

- When the thermal submodel runs at each time step and updates the temperature at each node, does the DFN automatically use this new temperature for calculations at the next time step?

Thank you again for the help!

**Comment by rtimms** [comment, marked as answer]
Looks like you have version 0.4.0, which is very old. I’d recommend updating. 

The thermal and electrochemical models are solved together, so the electrochemical equations are always using the temperature from the thermal submodel.

**Comment by agriyakhetarpal** [comment]
Version 0.4.0 as @rtimms mentioned did not have an upper bound on the Python version required back then, which makes `pip` pick it up on unsupported Python versions that came out later. I would recommend checking the Python version installed; PyBaMM 24.1 supports Python 3.8–3.12, but PyBaMM 23.5 and PyBaMM 23.9 support Python 3.8–3.11.

**Comment by abhishek-appana** [comment]
Got it!

Thank you @rtimms and @agriyakhetarpal 

Is there any quick way to update the codes from v0.4.0 to v24.1? Or do I need to manually figure out where the changes have been made and modify the codes accordingly?

**Reply by agriyakhetarpal** [reply]
Hi @abhishek-appana, I don't think there is a reliable or quick way to update your code snippets, I'm afraid. You shall have to browse the documentation to modify them based on changes to the PyBaMM core API.

**Reply by abhishek-appana** [reply]
Oh :-(

Okay. Thank you for the help. Appreciate it :-)
